.github/workflows/coverage.yml

```diff
@@ -0,0 +1,128 @@
+      - 'cmake/manifests/linux/**'
+      - '!container/**'
+      - '!docs/**'
+      - '!scripts/**'
+
+jobs:
+  build:
+    runs-on: ubuntu-20.04
+
+    container:
+      image: ghcr.io/intel/fpga-runtime-for-opencl/ubuntu-20.04-dev:main
+
+    continue-on-error: true
+
+    env:
+      WORK_DIR: /__w/fpga-runtime-for-opencl/fpga-runtime-for-opencl
+      TMP_DIR: /home/build/
+
+    steps:
+      - name: change ownership of workspace to current user
+        run: sudo chown -R build:build .
+
+      - name: checkout main branch
+        uses: actions/checkout@v2
+        with:
+          ref: ${{ github.base_ref }}
+      
+      - name: create build directories
+        run: mkdir "${{ env.TMP_DIR }}"/child_build && mkdir "${{ env.TMP_DIR }}"/parent_build
+
+      - name: create parent build files
+        run: cd "${{ env.TMP_DIR }}"/parent_build && cmake -G Ninja "${{ env.WORK_DIR }}" -DCMAKE_BUILD_TYPE=Debug -DCMAKE_INSTALL_PREFIX=install -DACL_CODE_COVERAGE=ON
+
+      - name: build parent runtime
+        run: cd "${{ env.TMP_DIR }}"/parent_build && ninja -v -j1 -k0
+
+      - name: parent runtime coverage scan
+        run: cd "${{ env.TMP_DIR }}"/parent_build && ctest -T Test -T Coverage
+
+      - name: checkout current branch
+        uses: actions/checkout@v2
+
+      - name: create child build files
+        run: cd "${{ env.TMP_DIR }}"/child_build && cmake -G Ninja "${{ env.WORK_DIR }}" -DCMAKE_BUILD_TYPE=Debug -DCMAKE_INSTALL_PREFIX=install -DACL_CODE_COVERAGE=ON
+
+      - name: build child runtime
+        run: cd "${{ env.TMP_DIR }}"/child_build && ninja -v -j1 -k0
+
+      - name: child runtime coverage scan
+        run: cd "${{ env.TMP_DIR }}"/child_build && ctest -T Test -T Coverage
+
+      - name: coverage status
+        continue-on-error: true
+        run: ./scripts/coverage_diff.py "${{ env.TMP_DIR }}"/child_build "${{ env.TMP_DIR }}"/parent_build || echo $? > "${{ env.TMP_DIR }}"/exit_code
+
+      - name: echo exit status
+        run: |
+          if [ -e "${{ env.TMP_DIR }}"/exit_code ]
+          then
+              cat "${{ env.TMP_DIR }}"/exit_code
+          else
+              echo 0
+              echo 0 > "${{ env.TMP_DIR }}"/exit_code
+          fi
+      - name: new file(s) covered
+        run: |
+          CODE=$(cat "${{ env.TMP_DIR }}"/exit_code)
+          if [ \'"${CODE}"\' = \'1\' ]; then
+            echo "New file must be covered"
+            exit 1
+          else
+            echo "New file covered."
+          fi
+      - name: no severe coverage decrease
+        run: |
+          CODE=$(cat "${{ env.TMP_DIR }}"/exit_code)
+          if [ \'"${CODE}"\' = \'2\' ]; then
+            echo "At least 1 file's coverage decrease by more than 20%"
+            exit 1
+          else
+            echo "No severe coverage decrease detected."
+          fi
+      - name: coverage non-decreasing
+        if: $(cat "${{ env.TMP_DIR }}"/exit_code) == '0'
+        run: echo "Coverage did not decrease."
```

Comment:
On first look the resulting output of the exit code handling is nice. On the other hand it complicates the workflow quite a bit. Would you agree with simply invoking the python script, and letting the workflow fail without further output when the python scripts exits with a non-zero code? You print all the error cases within the script, which is perfectly sufficient.

The larger issue with workflows is that the workflow code cannot be tested outside of that environment, so I much prefer your python script over the additional workflow pretty printing. It's a balance, sometimes an inline [one-liner like this](https://github.com/intel/fpga-runtime-for-opencl/blob/a2b6ff4abf1385a477872cc85cc936521208ba5f/.github/workflows/unix.yml#L26-L33) is ok, but in this case the flow is more complicated and can live in the manually testable script.